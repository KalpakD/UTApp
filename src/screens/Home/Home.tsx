import * as React from 'react';
import {View, Text, Button, TextInput, ScrollView} from 'react-native';
import crashlytics from '@react-native-firebase/crashlytics';
import {useNavigation} from '@react-navigation/native';

const Home = () =>
  // {navigation}: any
  {
    let navigation: any = useNavigation();
    let data = null;
    const [value, setValue] = React.useState(0);
    const [type, setType] = React.useState('');
    React.useEffect(() => {
      crashlytics().log('Home Screen mounted.');
    }, [data]);

    React.useEffect(() => {
      if (value) {
        if (value > 500) {
          setType('Expensive');
        } else if (value > 100) {
          setType('Not Expensive');
        } else {
          setType('Great');
        }
      }
    }, [value]);
    return (
      <ScrollView style={{flex: 1}}>
        <View
          style={{
            flex: 1,
            alignItems: 'center',
            justifyContent: 'center',
            marginTop: 10,
          }}>
          <Text testID="HomePress" onPress={() => crashlytics().crash()}>
            Home Screen
          </Text>

          <Button
            testID="reduxBtn"
            title="Redux Example"
            onPress={() => navigation.navigate('CounterEx')}
          />
          <Button
            testID="sagaBtn"
            title="ReduxSaga Example"
            onPress={() => navigation.navigate('SagaEx')}
          />
          <TextInput
            testID="InputField"
            placeholder="Enter a Value"
            keyboardType="number-pad"
            style={{
              borderColor: 'black',
              borderWidth: 1,
              margin: 16,
              minWidth: 80,
              padding: 8,
            }}
            onChangeText={val => {
              setValue(Number(val));
            }}
          />
          <Text>Value Entered is {type}</Text>
        </View>
      </ScrollView>
    );
  };
export default Home;
