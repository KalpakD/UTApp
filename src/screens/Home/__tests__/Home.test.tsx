/**
 * @format
 */

import 'react-native';
import React from 'react';
import {render, screen, fireEvent} from '@testing-library/react-native';
import Home from 'screens/Home/Home';

describe('Render Home Components correctly ', () => {
  //this is  Snapshot testing
  it('renders correctly', () => {
    render(<Home />);
    expect(screen.toJSON()).toMatchSnapshot();
  });
  test('on Press Events on Home Screen', () => {
    //Mocking Navigation method navigate
    let props = {navigation: {navigate: jest.fn()}};
    const {getByTestId} = render(<Home {...props} />);
    fireEvent(getByTestId('HomePress'), 'onPress');
    fireEvent(getByTestId('reduxBtn'), 'onPress');
    fireEvent(getByTestId('sagaBtn'), 'onPress');
    fireEvent(getByTestId('InputField'), 'onChangeText', 10);
    fireEvent(getByTestId('InputField'), 'onChangeText', 110);
    fireEvent(getByTestId('InputField'), 'onChangeText', 1000);
  });
});
