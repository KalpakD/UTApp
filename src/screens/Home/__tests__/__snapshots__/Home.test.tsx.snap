// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Render Home Components correctly  renders correctly 1`] = `
<RCTScrollView
  style={
    {
      "flex": 1,
    }
  }
>
  <View>
    <View
      style={
        {
          "alignItems": "center",
          "flex": 1,
          "justifyContent": "center",
          "marginTop": 10,
        }
      }
    >
      <Text
        onPress={[Function]}
        testID="HomePress"
      >
        Home Screen
      </Text>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="reduxBtn"
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            Redux Example
          </Text>
        </View>
      </View>
      <View
        accessibilityRole="button"
        accessibilityState={
          {
            "busy": undefined,
            "checked": undefined,
            "disabled": undefined,
            "expanded": undefined,
            "selected": undefined,
          }
        }
        accessibilityValue={
          {
            "max": undefined,
            "min": undefined,
            "now": undefined,
            "text": undefined,
          }
        }
        accessible={true}
        collapsable={false}
        focusable={true}
        onClick={[Function]}
        onResponderGrant={[Function]}
        onResponderMove={[Function]}
        onResponderRelease={[Function]}
        onResponderTerminate={[Function]}
        onResponderTerminationRequest={[Function]}
        onStartShouldSetResponder={[Function]}
        style={
          {
            "opacity": 1,
          }
        }
        testID="sagaBtn"
      >
        <View
          style={
            [
              {},
            ]
          }
        >
          <Text
            style={
              [
                {
                  "color": "#007AFF",
                  "fontSize": 18,
                  "margin": 8,
                  "textAlign": "center",
                },
              ]
            }
          >
            ReduxSaga Example
          </Text>
        </View>
      </View>
      <TextInput
        keyboardType="number-pad"
        onChangeText={[Function]}
        placeholder="Enter a Value"
        style={
          {
            "borderColor": "black",
            "borderWidth": 1,
            "margin": 16,
            "minWidth": 80,
            "padding": 8,
          }
        }
        testID="InputField"
      />
      <Text>
        Value Entered is 
      </Text>
    </View>
  </View>
</RCTScrollView>
`;
